helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -1278,12 +1275,10 @@ private void fireDataChangedEvents(final String path, Set<IZkDataListenerEntry>
             + listener.getDataListener() + " prefetch data: " + listener.isPrefetchData()) {
           @Override
           public void run() throws Exception {
-            // Reinstall watch before listener callbacks to check the znode status
```

Comment:
Can you add a comment like this? We reinstall watch based on the node has been deleted or not.

Reply:
updated!